File: 1.1/src/cz/mg/vulkan/PFNvkCreateFence.java
package cz.mg.vulkan;

public class PFNvkCreateFence extends VkFunctionPointer {
    public PFNvkCreateFence() {
    }

    protected PFNvkCreateFence(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCreateFence(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCreateFence(long value) {
        setValue(value);
    }

    public PFNvkCreateFence(VkInstance instance) {
        super(instance, new VkString("vkCreateFence"));
    }

    public int call(VkDevice device, VkFenceCreateInfo pCreateInfo, VkAllocationCallbacks pAllocator, VkFence pFence){
        return callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pCreateInfo != null ? pCreateInfo.getVkAddress() : VkPointer.NULL, pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL, pFence != null ? pFence.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callNative(long vkaddress, long device, long pCreateInfo, long pAllocator, long pFence);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkCreateFence.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkCreateFence_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pCreateInfo, jlong pAllocator, jlong pFence){
    (void)env;
    (void)jc;
    PFN_vkCreateFence f = (PFN_vkCreateFence)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkFenceCreateInfo*)jniLongToPointer(pCreateInfo)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator)),
        ((VkFence*)jniLongToPointer(pFence))
    );
}

